subsys/bluetooth/mesh/dfu_slot.h

```diff
@@ -16,48 +16,67 @@
 typedef enum bt_mesh_dfu_iter (*bt_mesh_dfu_slot_cb_t)(
 	const struct bt_mesh_dfu_slot *slot, void *user_data);
 
-/** @brief Register a new DFU image slot for a distributable image.
+/** @brief Get the number of slots committed to the firmware list.
+ *
+ *  @return Number of committed slots.
+ */
+int bt_mesh_dfu_slot_count(void);
+
+/** @brief Reserve a new DFU image slot for a distributable image.
  *
  *  A DFU image slot represents a single distributable DFU image with all its
- *  metadata.
+ *  metadata. The slot data must be set and @c bt_mesh_dfu_slot_commit must be
+ *  called for the slot to be considered part of the slot list.
  *
- *  @note The slot is allocated as invalid. Call
- *        @ref bt_mesh_dfu_slot_valid_set to make it valid.
+ *  @return A pointer to the reserved slot, or NULL if allocation failed.
+ */
+struct bt_mesh_dfu_slot *bt_mesh_dfu_slot_reserve(void);
+
+/** @brief Set the size and metadata for a reserved slot.
  *
- *  @param size         Size of the image in bytes.
- *  @param fwid         Firmware ID.
- *  @param fwid_len     Length of the firmware ID, at most @c
- *                      CONFIG_BT_MESH_DFU_FWID_MAXLEN.
+ *  @param dfu_slot     Pointer to the reserved slot for which to set the
+ *                      metadata.
+ *  @param size         The size of the image.
  *  @param metadata     Metadata or NULL.
  *  @param metadata_len Length of the metadata, at most @c
  *                      CONFIG_BT_MESH_DFU_METADATA_MAXLEN.
- *  @param uri          Image URI or NULL.
- *  @param uri_len      Length of the image URI, at most @c
- *                      CONFIG_BT_MESH_DFU_URI_MAXLEN.
  *
- *  @return A pointer to the allocated slot, or NULL if allocation failed.
+ *  @return 0 on success, (negative) error code otherwise.
  */
-const struct bt_mesh_dfu_slot *
-bt_mesh_dfu_slot_add(size_t size, const uint8_t *fwid, size_t fwid_len,
-		     const uint8_t *metadata, size_t metadata_len,
-		     const char *uri, size_t uri_len);
+int bt_mesh_dfu_slot_info_set(struct bt_mesh_dfu_slot *dfu_slot, size_t size,
+			      const uint8_t *metadata, size_t metadata_len);
 
-/** @brief Set whether the given slot is valid.
+/** @brief Set the fwid for a reserved slot.
  *
- *  @param slot  Allocated DFU image slot.
- *  @param valid New valid state of the slot.
+ *  @param dfu_slot Pointer to the reserved slot for which to set the fwid.
+ *  @param fwid     Metadata or NULL.
+ *  @param fwid_len Length of the fwid, at most @c
+ *                  CONFIG_BT_MESH_DFU_FWID_MAXLEN.
  *
- *  @return 0 on success, or (negative) error code on failure.
+ *  @return 0 on success, (negative) error code otherwise.
+ */
+int bt_mesh_dfu_slot_fwid_set(struct bt_mesh_dfu_slot *dfu_slot,
```

Comment:
Ahh right. Then, I think, we need to clarify this a bit. Because then "fwid" is the "new fwid" ... and application should not mix that with the "old fwid". Because this part is not explained in this API documentation. Slots are identified by the "fwid" contained in them, and not the "fwid" of the firmware on the chip or the remote node.

Reply:
Keep in mind that this file is an internal API, so no application will call these directly. But I'll clarify a bit.